### ISSUE TITLE
strange errors about proxy server

### ISSUE THREAD

**Issue by albertz** [openlierox/openlierox#195]
Converted from [SourceForge issue 2698107](http://sourceforge.net/support/tracker.php?aid=2698107), submitted by albertzeyer

Most strange is that I don't have a proxy set. This is the output:

W: Http: proxy "" failed, trying a direct connection
backtrace() returned 15 addresses
0   OpenLieroX                          0x002c8b9e _Z19DumpCallstackPrintfPv + 44
1   OpenLieroX                          0x002c9375 _Z13DumpCallstackPFvRKSsE + 907
2   OpenLieroX                          0x002c95ea _ZN6Logger5flushEv + 44
3   OpenLieroX                          0x0005a857 _Z15featureArrayLenv + 253
4   OpenLieroX                          0x000fd216 _Z4endlR6Logger + 62
5   OpenLieroX                          0x00125df7 _Z7GetTypeIiE15ScriptVarType_tv + 1429
6   OpenLieroX                          0x0014eba7 _ZN5CHttp15ProcessInternalEv + 703
7   OpenLieroX                          0x00153460 _ZSt9make_pairIP18HttpRetryEventDataP9SDL_mutexESt4pairIT_T0_ES5_S6_ + 5730
8   OpenLieroX                          0x001534e1 _ZSt9make_pairIP18HttpRetryEventDataP9SDL_mutexESt4pairIT_T0_ES5_S6_ + 5859
9   OpenLieroX                          0x002cdb85 _ZNSt4listI9SDL_EventSaIS0_EE6insertESt14_List_iteratorIS0_ERKS0_ + 207
10  OpenLieroX                          0x002ce8de _ZN10ThreadPool13threadWrapperEPv + 316
11  SDL                                 0x0331a14d SDL_Linked_Version + 397
12  SDL                                 0x03329341 SDL_SemWait + 49
13  libSystem.B.dylib                   0x92686095 _pthread_start + 321
14  libSystem.B.dylib                   0x92685f52 thread_start + 34
n: Registering server at lieroxtreme.thegaminguniverse.com/server
W: Http: proxy "" failed, trying a direct connection
backtrace() returned 15 addresses
0   OpenLieroX                          0x002c8b9e _Z19DumpCallstackPrintfPv + 44
1   OpenLieroX                          0x002c9375 _Z13DumpCallstackPFvRKSsE + 907
2   OpenLieroX                          0x002c95ea _ZN6Logger5flushEv + 44
3   OpenLieroX                          0x0005a857 _Z15featureArrayLenv + 253
4   OpenLieroX                          0x000fd216 _Z4endlR6Logger + 62
5   OpenLieroX                          0x00125df7 _Z7GetTypeIiE15ScriptVarType_tv + 1429
6   OpenLieroX                          0x0014eba7 _ZN5CHttp15ProcessInternalEv + 703
7   OpenLieroX                          0x00153460 _ZSt9make_pairIP18HttpRetryEventDataP9SDL_mutexESt4pairIT_T0_ES5_S6_ + 5730
8   OpenLieroX                          0x001534e1 _ZSt9make_pairIP18HttpRetryEventDataP9SDL_mutexESt4pairIT_T0_ES5_S6_ + 5859
9   OpenLieroX                          0x002cdb85 _ZNSt4listI9SDL_EventSaIS0_EE6insertESt14_List_iteratorIS0_ERKS0_ + 207
10  OpenLieroX                          0x002ce8de _ZN10ThreadPool13threadWrapperEPv + 316
11  SDL                                 0x0331a14d SDL_Linked_Version + 397
12  SDL                                 0x03329341 SDL_SemWait + 49
13  libSystem.B.dylib                   0x92686095 _pthread_start + 321
14  libSystem.B.dylib                   0x92685f52 thread_start + 34
E: OpenReliableSocket: Bad file descriptor.
backtrace() returned 26 addresses
0   OpenLieroX                          0x002c8b9e _Z19DumpCallstackPrintfPv + 44
1   OpenLieroX                          0x002c9375 _Z13DumpCallstackPFvRKSsE + 907
2   OpenLieroX                          0x002c95ea _ZN6Logger5flushEv + 44
3   OpenLieroX                          0x0005a857 _Z15featureArrayLenv + 253
4   OpenLieroX                          0x000fd216 _Z4endlR6Logger + 62
5   OpenLieroX                          0x00125df7 _Z7GetTypeIiE15ScriptVarType_tv + 1429
6   OpenLieroX                          0x0010c0c3 _Z18OpenReliableSockett + 285
7   OpenLieroX                          0x00148a90 _ZN5CHttp12InitTransferERKSsS1_ + 170
8   OpenLieroX                          0x00148cfd _ZN5CHttp11RequestDataERKSsS1_ + 37
9   OpenLieroX                          0x00148d90 _ZN5CHttp16RetryWithNoProxyERKSsS1_ + 124
10  OpenLieroX                          0x00148f8b _ZN5CHttp18HttpThread_onRetryE12SmartPointerI18HttpRetryEventData10NopFunctorIPvEE + 71
11  OpenLieroX                          0x00153f32 _ZSt9make_pairIP18HttpRetryEventDataP9SDL_mutexESt4pairIT_T0_ES5_S6_ + 8500
12  OpenLieroX                          0x001543fc _Z10isSameTypeIN5EventIN5CHttp13HttpEventDataEE12Handler_NoOpENS3_7HandlerEEbRKT0_RKT_ + 212
13  OpenLieroX                          0x00154467 _Z10isSameTypeIN5EventIN5CHttp13HttpEventDataEE12Handler_NoOpENS3_7HandlerEEbRKT0_RKT_ + 319
14  OpenLieroX                          0x000f7f98 _Z25WasKeyboardEventHappeningib + 692
15  OpenLieroX                          0x000fb7d6 _ZNSt6vectorI10MouseEventSaIS0_EE13_M_insert_auxEN9__gnu_cxx17__normal_iteratorIPS0_S2_EERKS0_ + 1400
16  OpenLieroX                          0x000f945e _Z15InitEventSystemv + 1666
17  OpenLieroX                          0x000f82d6 _Z15HandleNextEventv + 84
18  OpenLieroX                          0x000f8c4b _Z13ProcessEventsv + 35
19  OpenLieroX                          0x00103d5c _Z26SetCrashHandlerReturnPointPKc + 2140
20  OpenLieroX                          0x002cdb85 _ZNSt4listI9SDL_EventSaIS0_EE6insertESt14_List_iteratorIS0_ERKS0_ + 207
21  OpenLieroX                          0x002ce8de _ZN10ThreadPool13threadWrapperEPv + 316
22  SDL                                 0x0331a14d SDL_Linked_Version + 397
23  SDL                                 0x03329341 SDL_SemWait + 49
24  libSystem.B.dylib                   0x92686095 _pthread_start + 321
25  libSystem.B.dylib                   0x92685f52 thread_start + 34
n: Registering on UDP masterserver server.az2000.de:23450
n: GameServer::ParseServerRegistered()

**Comment by albertz** [comment]
Submitted by albertzeyer

Full output:

[Session started at 2009-03-20 19:00:58 +0100.]
H: OpenLieroX/0.57_beta9 is starting ...
H: This is a DEBUG build.
n: ThreadPool: creating 40 threads ...
n: Loading options...
n: I have now the following searchpaths (in this order):
n:   /Users/az/Library/Application Support/OpenLieroX
n:   /Users/az/Programmierung/openlierox/share/gamedir
n:   .
n:   /Users/az/Programmierung/openlierox/build/Xcode/build/Debug/OpenLieroX.app/Contents/MacOS/../Resources/gamedir
n:   /usr/share/OpenLieroX
n:  And that's all.
n: Reading game options from /Users/az/Library/Application Support/OpenLieroX/cfg/options.cfg
n: Will write game options to /Users/az/Library/Application Support/OpenLieroX/cfg/options.cfg
n: the option "Advanced.AllowDirtUpdates" defined in options.cfg is unknown
n: the option "Advanced.AllowFileDownload" defined in options.cfg is unknown
n: the option "Advanced.CountTeamkills" defined in options.cfg is unknown
n: the option "Advanced.NetworkUpdatePeriod" defined in options.cfg is unknown
n: the option "Advanced.ServerSideHealth" defined in options.cfg is unknown
n: the option "Advanced.WeaponSelectionMaxTime" defined in options.cfg is unknown
n: the option "Game.AllowMouseAiming" defined in options.cfg is unknown
n: the option "Game.AllowStrafing" defined in options.cfg is unknown
n: the option "Game.UseNumericKeysToSwitchWeapons" defined in options.cfg is unknown
n: the option "LastGame.AllowNickChange" defined in options.cfg is unknown
n: the option "LastGame.AllowRemoteBots" defined in options.cfg is unknown
n: the option "LastGame.AllowWantsJoinMsg" defined in options.cfg is unknown
n: the option "LastGame.BonusFrequency" defined in options.cfg is unknown
n: the option "LastGame.BonusHealthToWeaponChance" defined in options.cfg is unknown
n: the option "LastGame.BonusLife" defined in options.cfg is unknown
n: the option "LastGame.BonusNames" defined in options.cfg is unknown
n: the option "LastGame.Bonuses" defined in options.cfg is unknown
n: the option "LastGame.EmptyWeaponsOnRespawn" defined in options.cfg is unknown
n: the option "LastGame.GameType" defined in options.cfg is unknown
n: the option "LastGame.GroupTeamScore" defined in options.cfg is unknown
n: the option "LastGame.KillLimit" defined in options.cfg is unknown
n: the option "LastGame.LastSelectedPlayer" defined in options.cfg is unknown
n: the option "LastGame.LevelName" defined in options.cfg is unknown
n: the option "LastGame.Lives" defined in options.cfg is unknown
n: the option "LastGame.LoadingTime" defined in options.cfg is unknown
n: the option "LastGame.MatchLogging" defined in options.cfg is unknown
n: the option "LastGame.MaxPlayers" defined in options.cfg is unknown
n: the option "LastGame.ModName" defined in options.cfg is unknown
n: the option "LastGame.Password" defined in options.cfg is unknown
n: the option "LastGame.RegisterServer" defined in options.cfg is unknown
n: the option "LastGame.RespawnGroupTeams" defined in options.cfg is unknown
n: the option "LastGame.RespawnInWaves" defined in options.cfg is unknown
n: the option "LastGame.RespawnTime" defined in options.cfg is unknown
n: the option "LastGame.ServerName" defined in options.cfg is unknown
n: the option "LastGame.SuicideDecreasesScore" defined in options.cfg is unknown
n: the option "LastGame.TagLimit" defined in options.cfg is unknown
n: the option "LastGame.TimeLimit" defined in options.cfg is unknown
n: the option "LastGame.TopBarVisible" defined in options.cfg is unknown
n: the option "LastGame.WantsToJoinFromBanned" defined in options.cfg is unknown
n: the option "LastGame.WelcomeMessage" defined in options.cfg is unknown
n: the option "MediaPlayer.Left" defined in options.cfg is unknown
n: the option "MediaPlayer.MusicVolume" defined in options.cfg is unknown
n: the option "MediaPlayer.Repeat" defined in options.cfg is unknown
n: the option "MediaPlayer.Shuffle" defined in options.cfg is unknown
n: the option "MediaPlayer.Top" defined in options.cfg is unknown
n: the option "Network.LoadDbAtStartup" defined in options.cfg is unknown
n: the option "Network.NatTraverse" defined in options.cfg is unknown
n: the option "Network.SendServerSideFeatures" defined in options.cfg is unknown
n: Default theme is used
H: Unknown options were found.
n: DONE loading options
n: Installing CrashHandler .. no signal handler with these settings
n: Loading network texts... DONE
n: Hello there, I am initializing me now...
H: DEDICATED MODE
n: soundsystem completly disabled
H: SmartPointer collision detector initialized
n: Loading: Initializing network
n: Loading: Initializing client and server
n: Loading: Initializing game entities
n: Loading: Loading graphics
W: could not load image data/frontend/speedtest.png
backtrace() returned 32 addresses
0   OpenLieroX                          0x002c8b9e _Z19DumpCallstackPrintfPv + 44
1   OpenLieroX                          0x002c9375 _Z13DumpCallstackPFvRKSsE + 907
2   OpenLieroX                          0x002c95ea _ZN6Logger5flushEv + 44
3   OpenLieroX                          0x0005a857 _Z15featureArrayLenv + 253
4   OpenLieroX                          0x000fd216 _Z4endlR6Logger + 62
5   OpenLieroX                          0x00125df7 _Z7GetTypeIiE15ScriptVarType_tv + 1429
6   OpenLieroX                          0x00045c9b _Z20Load_Image_WithAlphaR12SmartPointerI11SDL_Surface10NopFunctorIPvEERKSs + 209
7   OpenLieroX                          0x001cb5e0 _ZN13DeprecatedGUI12LoadGraphicsEv + 1052
8   OpenLieroX                          0x00106634 _Z16InitializeLieroXv + 2798
9   OpenLieroX                          0x0010745d SDL_main + 1395
10  OpenLieroX                          0x000036ab start + 2843
11  Foundation                          0x93e51e1a NSZoneMalloc + 1770
12  CoreFoundation                      0x92b998da _CFXNotificationResetSessionForTask + 1514
13  CoreFoundation                      0x92b99bb3 _CFXNotificationPostNotification + 179
14  Foundation                          0x93e4f080 NSPopAutoreleasePool + 5088
15  Foundation                          0x93e588c8 _NSDescriptionWithLocaleFunc + 5464
16  AppKit                              0x90f9049a _NSSubclassOverridesSelector + 8449
17  AppKit                              0x90f903a9 _NSSubclassOverridesSelector + 8208
18  AppKit                              0x90f09ec3 _NSInitializeKillRing + 6437
19  AppKit                              0x90f096bc _NSInitializeKillRing + 4382
20  Foundation                          0x93e7743f NSZoneRealloc + 3199
21  Foundation                          0x93e7714f NSZoneRealloc + 2447
22  AE                                  0x924cf648 aeProcessAppleEvent + 724
23  AE                                  0x924cf57e aeProcessAppleEvent + 522
24  AE                                  0x924cf425 aeProcessAppleEvent + 177
25  HIToolbox                           0x907299a5 AEProcessAppleEvent + 38
26  AppKit                              0x90f06f91 _DPSNextEvent + 1189
27  AppKit                              0x90f06630 _NSUpdateMenuRefWithChangedMenuItem + 2250
28  AppKit                              0x90eff66b _NSSetViewMultiClipDrawingHelper + 4866
29  OpenLieroX                          0x000034ad start + 2333
30  OpenLieroX                          0x0000396e main + 305
31  OpenLieroX                          0x00002bc6 start + 54
n: Loading: Initializing console
n: Loading: Loading sounds
n: Loading: Loading player profiles
n: Dedicated server: running script "/Users/az/Programmierung/openlierox/share/gamedir/scripts/dedicated_control"
n: Initializing ready
n: Loading: Initializing menu system
n: Loading: Loading Physics Engine
n: PhysicsEngine LX56 physics loaded
n: Loading: Done! Starting menu
n: Server max upload bandwidth is 832902 bytes/s
H: server started on 134.130.188.194:23400
n: Registering server at thelobby.altervista.org/server
n: sending challenge request to 127.0.0.1:23400
n: Server is using OpenLieroX/0.57_beta9
n: CClient: got challenge response from OpenLieroX/0.57_beta9 server
n: GameServer: our local client has connected
n: CheckForFillWithBots: adding 4 bots
H: Worm added: [CPU] Dummi (id 0)
H: Worm added: [CPU] Kamikazee! (id 1)
H: Worm added: The Third (id 2)
H: Worm added: [CPU] K1ll0r (id 3)
n: Registering server at lieroxtreme.thegaminguniverse.com/server
H: DedicatedControl: message: Preset Mortars
H: DedicatedControl: message: 2009-03-20 19:01:01 -- INFO -- Dedicated_control started
H: DedicatedControl: message: Preset MSF_II
n: Registering on UDP masterserver server.az2000.de:23450
n: WARNING: Version::setByString: '0.57_beta4' get parsed as 'OpenLieroX/0.57_beta4'
H: Worm joined: hupp (id 4, from 88.194.239.138:1082(OpenLieroX/0.57_beta4))
n: CheckForFillWithBots: removing 1 bots
n: Worm was kicked (too much players, bot not needed anymore): [CPU] K1ll0r (id 3)
H: Worm left: [CPU] K1ll0r (id 3)
n: Current cache size: 5 MB
n: Registering on UDP masterserver server.az2000.de:23450
n: GameServer::ParseServerRegistered()
H: Worm left: hupp (id 4)
n: CheckForFillWithBots: adding 1 bots
H: Worm added: [CPU] Worm (id 3)
n: GameServer::StartGame() mod MSF II
n: CMap::Load: level levels/GammaComplex.lxl is in image format
n: LoadTheme: Theme dirt already loaded
n: Map loadtime: 0.295 seconds
n: modLog: Loading game mod file MSF II/script.lgs
n: Mod loadtime: 0.208 seconds
n: Registering on UDP masterserver server.az2000.de:23450
n: MaxFPS is 95
n: GameLoopStart
n: GameServer::ParseServerRegistered()
n: Client: Got ParsePrepareGame
n: preparing worm 0:[CPU] Dummi for battle
n: preparing worm 1:[CPU] Kamikazee! for battle
n: preparing worm 2:The Third for battle
n: preparing worm 3:[CPU] Worm for battle
H: Client: we are ready, waiting now for start game signal
n: getting weapons for worm (0) [CPU] Dummi
n: getting weapons for worm (1) [CPU] Kamikazee!
n: getting weapons for worm (2) The Third
n: getting weapons for worm (3) [CPU] Worm
H: Server: BeginMatch
n: Server 'OpenLieroX Server' game state:
n:  \* playing, Team Death Match
n:  \* maxkills=-1, lives=-2, timelimit=-60 (curtime=0)
n:  + 0:'[CPU] Dummi'(bot), team 1, alive, not yet prepared, lives=-2, kills=0 on local(OpenLieroX/0.57_beta9)
n:  + 1:'[CPU] Kamikazee!'(bot), team 0, alive, not yet prepared, lives=-2, kills=0 on local(OpenLieroX/0.57_beta9)
n:  + 2:'The Third'(bot), team 1, alive, not yet prepared, lives=-2, kills=0 on local(OpenLieroX/0.57_beta9)
n:  + 3:'[CPU] Worm'(bot), team 0, alive, not yet prepared, lives=-2, kills=0 on local(OpenLieroX/0.57_beta9)
n: preparing game mode Team Death Match
n: Registering on UDP masterserver server.az2000.de:23450
n: GameServer::ParseServerRegistered()
n: getting weapons for worm (0) [CPU] Dummi
n: getting weapons for worm (1) [CPU] Kamikazee!
n: getting weapons for worm (2) The Third
n: getting weapons for worm (3) [CPU] Worm
n: Client: get start game signal
n: Current cache size: 11 MB
H: Worm joined: P.L.G (id 4, from 88.196.98.136:53899(OpenLieroX/0.57_beta8))
n: CheckForFillWithBots: removing 1 bots
n: Worm was kicked (too much players, bot not needed anymore): [CPU] Worm (id 3)
n: using max upload rate 813.381 kb/sec
H: Worm left: [CPU] Worm (id 3)
n: getting weapons for worm (4) P.L.G
H: Server: BeginMatch for 88.196.98.136:53899(OpenLieroX/0.57_beta8)(4 'P.L.G')
n: getting weapons for worm (4) P.L.G
n: ParseConnect: 89.74.184.240:1092(OpenLieroX/0.57_beta5): OpenLieroX/0.57_beta5 is too old: connecting during game is allowed
n: Registering on UDP masterserver server.az2000.de:23450
n: GameServer::ParseServerRegistered()
n: ParseConnect: 89.74.184.240:1095(OpenLieroX/0.57_beta5): OpenLieroX/0.57_beta5 is too old: connecting during game is allowed
n: Registering on UDP masterserver server.az2000.de:23450
n: GameServer::ParseServerRegistered()
n: Registering on UDP masterserver server.az2000.de:23450
n: GameServer::ParseServerRegistered()
H: Worm joined: [Sun~]! Wertigo (id 3, from 213.29.61.125:24029(OpenLieroX/0.57_beta9_r3427))
n: CheckForFillWithBots: removing 1 bots
n: Worm was kicked (too much players, bot not needed anymore): The Third (id 2)
H: Worm left: The Third (id 2)
Traceback (most recent call last):
  File "/Users/az/Programmierung/openlierox/share/gamedir/scripts/dedicated_control_handler.py", line 142, in signalHandler
    parseWormDied(sig)
  File "/Users/az/Programmierung/openlierox/share/gamedir/scripts/dedicated_control_handler.py", line 319, in parseWormDied
    f.write( time.strftime("%Y-%m-%d %H:%M:%S") + "\t" + worms[deaderID].Name + "\t" + worms[killerID].Name + "\n" )
KeyError: 2
H: Worm left: [Sun~]! Wertigo (id 3)
n: CheckForFillWithBots: adding 1 bots
H: Worm added: [CPU] N00b (id 2)
n: Registering on UDP masterserver server.az2000.de:23450
n: GameServer::ParseServerRegistered()
n: WARNING: Version::setByString: '' get parsed as 'LieroX/0.56'
n: ParseConnect: 212.149.155.96:1034(LieroX/0.56): LieroX/0.56 is too old: connecting during game is allowed
n: Registering on UDP masterserver server.az2000.de:23450
n: GameServer::ParseServerRegistered()
n: Registering server at thelobby.altervista.org/server
n: Registering on UDP masterserver server.az2000.de:23450
n: GameServer::ParseServerRegistered()
W: Http: proxy "" failed, trying a direct connection
backtrace() returned 15 addresses
0   OpenLieroX                          0x002c8b9e _Z19DumpCallstackPrintfPv + 44
1   OpenLieroX                          0x002c9375 _Z13DumpCallstackPFvRKSsE + 907
2   OpenLieroX                          0x002c95ea _ZN6Logger5flushEv + 44
3   OpenLieroX                          0x0005a857 _Z15featureArrayLenv + 253
4   OpenLieroX                          0x000fd216 _Z4endlR6Logger + 62
5   OpenLieroX                          0x00125df7 _Z7GetTypeIiE15ScriptVarType_tv + 1429
6   OpenLieroX                          0x0014eba7 _ZN5CHttp15ProcessInternalEv + 703
7   OpenLieroX                          0x00153460 _ZSt9make_pairIP18HttpRetryEventDataP9SDL_mutexESt4pairIT_T0_ES5_S6_ + 5730
8   OpenLieroX                          0x001534e1 _ZSt9make_pairIP18HttpRetryEventDataP9SDL_mutexESt4pairIT_T0_ES5_S6_ + 5859
9   OpenLieroX                          0x002cdb85 _ZNSt4listI9SDL_EventSaIS0_EE6insertESt14_List_iteratorIS0_ERKS0_ + 207
10  OpenLieroX                          0x002ce8de _ZN10ThreadPool13threadWrapperEPv + 316
11  SDL                                 0x0331a14d SDL_Linked_Version + 397
12  SDL                                 0x03329341 SDL_SemWait + 49
13  libSystem.B.dylib                   0x92686095 _pthread_start + 321
14  libSystem.B.dylib                   0x92685f52 thread_start + 34
n: Registering server at lieroxtreme.thegaminguniverse.com/server
W: Http: proxy "" failed, trying a direct connection
backtrace() returned 15 addresses
0   OpenLieroX                          0x002c8b9e _Z19DumpCallstackPrintfPv + 44
1   OpenLieroX                          0x002c9375 _Z13DumpCallstackPFvRKSsE + 907
2   OpenLieroX                          0x002c95ea _ZN6Logger5flushEv + 44
3   OpenLieroX                          0x0005a857 _Z15featureArrayLenv + 253
4   OpenLieroX                          0x000fd216 _Z4endlR6Logger + 62
5   OpenLieroX                          0x00125df7 _Z7GetTypeIiE15ScriptVarType_tv + 1429
6   OpenLieroX                          0x0014eba7 _ZN5CHttp15ProcessInternalEv + 703
7   OpenLieroX                          0x00153460 _ZSt9make_pairIP18HttpRetryEventDataP9SDL_mutexESt4pairIT_T0_ES5_S6_ + 5730
8   OpenLieroX                          0x001534e1 _ZSt9make_pairIP18HttpRetryEventDataP9SDL_mutexESt4pairIT_T0_ES5_S6_ + 5859
9   OpenLieroX                          0x002cdb85 _ZNSt4listI9SDL_EventSaIS0_EE6insertESt14_List_iteratorIS0_ERKS0_ + 207
10  OpenLieroX                          0x002ce8de _ZN10ThreadPool13threadWrapperEPv + 316
11  SDL                                 0x0331a14d SDL_Linked_Version + 397
12  SDL                                 0x03329341 SDL_SemWait + 49
13  libSystem.B.dylib                   0x92686095 _pthread_start + 321
14  libSystem.B.dylib                   0x92685f52 thread_start + 34
E: OpenReliableSocket: Bad file descriptor.
backtrace() returned 26 addresses
0   OpenLieroX                          0x002c8b9e _Z19DumpCallstackPrintfPv + 44
1   OpenLieroX                          0x002c9375 _Z13DumpCallstackPFvRKSsE + 907
2   OpenLieroX                          0x002c95ea _ZN6Logger5flushEv + 44
3   OpenLieroX                          0x0005a857 _Z15featureArrayLenv + 253
4   OpenLieroX                          0x000fd216 _Z4endlR6Logger + 62
5   OpenLieroX                          0x00125df7 _Z7GetTypeIiE15ScriptVarType_tv + 1429
6   OpenLieroX                          0x0010c0c3 _Z18OpenReliableSockett + 285
7   OpenLieroX                          0x00148a90 _ZN5CHttp12InitTransferERKSsS1_ + 170
8   OpenLieroX                          0x00148cfd _ZN5CHttp11RequestDataERKSsS1_ + 37
9   OpenLieroX                          0x00148d90 _ZN5CHttp16RetryWithNoProxyERKSsS1_ + 124
10  OpenLieroX                          0x00148f8b _ZN5CHttp18HttpThread_onRetryE12SmartPointerI18HttpRetryEventData10NopFunctorIPvEE + 71
11  OpenLieroX                          0x00153f32 _ZSt9make_pairIP18HttpRetryEventDataP9SDL_mutexESt4pairIT_T0_ES5_S6_ + 8500
12  OpenLieroX                          0x001543fc _Z10isSameTypeIN5EventIN5CHttp13HttpEventDataEE12Handler_NoOpENS3_7HandlerEEbRKT0_RKT_ + 212
13  OpenLieroX                          0x00154467 _Z10isSameTypeIN5EventIN5CHttp13HttpEventDataEE12Handler_NoOpENS3_7HandlerEEbRKT0_RKT_ + 319
14  OpenLieroX                          0x000f7f98 _Z25WasKeyboardEventHappeningib + 692
15  OpenLieroX                          0x000fb7d6 _ZNSt6vectorI10MouseEventSaIS0_EE13_M_insert_auxEN9__gnu_cxx17__normal_iteratorIPS0_S2_EERKS0_ + 1400
16  OpenLieroX                          0x000f945e _Z15InitEventSystemv + 1666
17  OpenLieroX                          0x000f82d6 _Z15HandleNextEventv + 84
18  OpenLieroX                          0x000f8c4b _Z13ProcessEventsv + 35
19  OpenLieroX                          0x00103d5c _Z26SetCrashHandlerReturnPointPKc + 2140
20  OpenLieroX                          0x002cdb85 _ZNSt4listI9SDL_EventSaIS0_EE6insertESt14_List_iteratorIS0_ERKS0_ + 207
21  OpenLieroX                          0x002ce8de _ZN10ThreadPool13threadWrapperEPv + 316
22  SDL                                 0x0331a14d SDL_Linked_Version + 397
23  SDL                                 0x03329341 SDL_SemWait + 49
24  libSystem.B.dylib                   0x92686095 _pthread_start + 321
25  libSystem.B.dylib                   0x92685f52 thread_start + 34
n: Registering on UDP masterserver server.az2000.de:23450
n: GameServer::ParseServerRegistered()

**Comment by albertz** [comment]
Submitted by dark_charlie

Could you find out what does AutoSetupHttpProxy set the proxy to? I guess the bug is there because it does not happen on Windows.

**Comment by albertz** [comment]
Submitted by albertzeyer

It does not set any proxy (it's just the empty string).

**Comment by albertz** [comment]
Submitted by albertzeyer

The entry in the options.cfg about the proxy is just empty. I have also already deleted the entry multiple times, makes no difference.

Automatic detection is activated. Note that I had this bug already when there was just no code in the AutomaticProxyDetection for MacOSX! Now I also used the Linux case for MacOSX (no idea if MacOSX has special ways of checking that) but that doesn't change anything as I don't have these variables set.

I have this error both on my Linux server and MacOSX (didn't tried yet somewhere else).
